core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -500,6 +505,28 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
         }
     }
 
+    static void checkForIndexDataInDefaultPathData(final NodeEnvironment nodeEnv, final Logger logger) throws IOException {
+        if (nodeEnv.defaultNodePath() == null) {
+            return;
+        }
+        final Set<String> availableIndexFolders = nodeEnv.availableIndexFoldersForPath(nodeEnv.defaultNodePath());
+        if (!availableIndexFolders.isEmpty()) {
```

Comment:
I dislike checking size when we really only care about whether or not it's empty.

I pushed 9503f6fbb77f905e9fb007136aaa0622baf48d6c though, what do you think?